packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/FunctionsTests.kt

```diff
@@ -0,0 +1,630 @@
+//        val input = "Realm"
+//        val output = "Custom Realm"
+//        assertEquals(output, functions.callFunction(FIRST_ARG_FUNCTION, listOf(input), CustomStringDecoder(output)))
+//    }
+//
+//    @Test
+//    fun asyncResultDecoder() = looperThread.runBlocking {
+//        val input = "Realm"
+//        val output = "Custom Realm"
+//        functions.callFunctionAsync(FIRST_ARG_FUNCTION, listOf(input), CustomStringDecoder(output), App.Callback<String> { result ->
+//            try {
+//                assertEquals(output, result.orThrow)
+//            } finally {
+//                looperThread.testComplete()
+//            }
+//        })
+//    }
+
+    @Test
+    fun unknownFunction() {
+        assertFailsWithMessage<ServiceException>("[Service][FunctionNotFound(26)] function not found: 'unknown'") {
+            runBlocking {
+                functions.call<String>("unknown", listOf(32))
+            }
+        }
+    }
+
+//    @Test
+//    fun asyncUnknownFunction() = looperThread.runBlocking {
+//        val input = Dog("PojoFido")
+//        functions.callFunctionAsync("unknown", listOf(input), Dog::class.java, pojoRegistry) { result ->
+//            try {
+//                assertEquals(ErrorCode.FUNCTION_NOT_FOUND, result.error.errorCode)
+//            } finally {
+//                looperThread.testComplete()
+//            }
+//        }
+//    }
+//
+//    @Test
+//    fun asyncNonLoopers() {
+//        assertFailsWith<IllegalStateException> {
+//            functions.callFunctionAsync(FIRST_ARG_FUNCTION, listOf(32), Integer::class.java, pojoRegistry) { result ->
+//                fail()
+//            }
+//        }
+//    }
+
+    @Test
+    fun callFunction_sum() {
+        runBlocking {
+            assertEquals(10, functions.call<Int>(SUM_FUNCTION.name, 1, 2, 3, 4))
+        }
+    }
+
+    @Test
+    fun callFunction_remoteError() {
+        assertFailsWithMessage<ServiceException>("ReferenceError: 'unknown' is not defined") {
+            runBlocking {
+                functions.call<String>(ERROR_FUNCTION.name)
+            }
+        }
+    }
+
+    @Test
+    fun callFunction_null() {
+        runBlocking {
+            assertTrue(functions.call<BsonNull>(NULL_FUNCTION.name, emptyList<Any>()).isNull())
+        }
+    }
+
+    @Test
+    fun callFunction_void() {
+        runBlocking {
+            assertEquals(BsonType.UNDEFINED, functions.call<BsonUndefined>(VOID_FUNCTION.name).bsonType)
+        }
+    }
+
+    @Test
+    fun callFunction_afterLogout() {
+        runBlocking {
+            anonUser.logOut()
+        }
+        assertFailsWithMessage<ServiceException>("[Service][Unknown(-1)] expected Authorization header with JWT") {
+            runBlocking {
+                functions.call(FIRST_ARG_FUNCTION.name, 1, 2, 3)
+            }
+        }
+    }
+
+    // Tests that functions that should not execute based on "canevalute"-expression fails.
+    @Test
+    fun callFunction_authorizedOnly() {
```

Comment:
Given the error codes I posted earlier, I suspect we are missing some edge cases here like calling a non-existent function, wrong parameters and things like that. 

Also what happens here if the function throws an exception on the server